generators/app/templates/resources/lib/utilities.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+import xbmc
+import xbmcaddon
+import re
+import sys
+import logging
+
+<%_ if (props.kodiVersion == '2.24.0') { -%>
+if sys.version_info >= (2, 7):
+    import json as json
+else:
+    import simplejson as json
+<%_ } else { -%>
+import json as json
+<% } %>
+
+# read settings
+ADDON = xbmcaddon.Addon('<%= props.scriptid %>')
```

Comment:
Using `xbmcaddon.Addon()`is enough, no need to pass it an id. This is something that has been pointed out several times lately while submissions to the repository are reviewed